IoTuring/Configurator/ConfiguratorIO.py

```diff
@@ -0,0 +1,149 @@
+                    folderPath = os.path.join(folderPath, self.directoryName)
+        except:
+            pass # default folder path will be used
+        
+        # add slash if missing (for log reasons)
+        if not folderPath.endswith(os.sep):
+            folderPath += os.sep
+            
+        return folderPath
+
+    def defaultFolderPath(self):
+        return os.path.dirname(inspect.getfile(ConfiguratorIO))
+    
+    # https://developer.apple.com/library/archive/documentation/FileManagement/Conceptual/FileSystemProgrammingGuide/MacOSXDirectories/MacOSXDirectories.html
+    def macOSFolderPath(self):
+        paths = NSSearchPathForDirectoriesInDomains(NSApplicationSupportDirectory,NSUserDomainMask,True)
+        basePath = (len(paths) > 0 and paths[0]) or NSTemporaryDirectory()
+        return basePath
+    
+    # https://docs.microsoft.com/en-us/windows/win32/shell/knownfolderid
+    def windowsFolderPath(self):
+        return os.environ["APPDATA"]
+    
+    # https://specifications.freedesktop.org/basedir-spec/basedir-spec-latest.html
+    def linuxFolderPath(self):
+        return os.environ["XDG_CONFIG_HOME"] if "XDG_CONFIG_HOME" in os.environ else os.path.join(os.environ["HOME"], ".config")
+    
+    def envvarFolderPath(self):
+        return os.getenv(CONFIG_PATH_ENV_VAR)
+    
+    # In versions prior to 2022.12.2, the configurations file was stored in the same folder as this file
+    def oldFolderPath(self):
+        return os.path.dirname(inspect.getfile(ConfiguratorIO))
+    
+    def checkConfigurationFileInOldLocation(self):
+        """ Should be called at the beginning of the program when in -c mode.
+            Checks if a config. file is in the old folder path; if so ask if user wants to copy it
+            into new location, instead of rewriting it from zero.
+            If yes, copy it; otherwise leave it there and write a don't move file to remmeber the 
+            choose in the next times. 
+            This check is not done if old path = current chosen path 
+        """
+        
+        # This check is not done if old path = current chosen path (also check if ending slash is present)
+        if self.oldFolderPath() == self.getFolderPath() or self.oldFolderPath() == self.getFolderPath()[:-1]:
+            return
+        
+        # Exit check if no config. in old directory or if there is also the dont move file with it.
+        configuration_file_exists = os.path.exists(os.path.join(self.oldFolderPath(), CONFIGURATION_FILE_NAME))
+        dont_move_file_exists = os.path.exists(os.path.join(self.oldFolderPath(), DONT_MOVE_FILE_FILENAME))
+        if not configuration_file_exists or (configuration_file_exists and dont_move_file_exists):
+            return
+        
+        response = input("A configuration file has been found in the old location. Do you want to move it to the new location or to use a new blank configuration ? (y/n): ")
```

Comment:
This question is a bit confusing. You have to select yes/no but the question asks `Do you want to move it to the new location or to use a new blank configuration`, which is not a yes no question.